### ISSUE TITLE
Uplift Model Prediction Returns The Exact Same Result for All Rows in Test Set

### ISSUE THREAD

**Issue by lleiou** [uber/causalml#298]
**Describe the bug**
Hello, I'm building an uplift tree model using `UpliftRandomForestClassifier` following [uplift_trees_with_synthetic_data.ipynb](https://github.com/uber/causalml/blob/master/examples/uplift_trees_with_synthetic_data.ipynb). The data has 3 features and has 2 treatments and 1 control, the data is pretty imbalanced (less than 0.2% positive). The issue that I am experiencing right now is that the predictions for each row on the testing set are exactly the same despite the different values of features each row has. Would like to know if this is caused by data imbalance or the fact that the amount of feature I have is too small or is there any other parameters that I should set (or any other possible explanations)? Thanks!

**To Reproduce**
not sure if it could be reproduced because I cannot share the data, but here is basically how I coded it:
```python
from causalml.inference.tree import UpliftRandomForestClassifier
uplift_model = UpliftRandomForestClassifier(control_name='control')
uplift_model.fit(df_train[x_names].values,
                 treatment=df_train['treatment_group_key'].values,
                 y=df_train['conversion'].values)
y_pred = uplift_model.predict(df_test[x_names].values)
```

**Expected behavior**
The predicted lift values should be different for records with different feature values.

**Comment by lleiou** [comment]
Reading data from `.parquet` files using pandas caused feature columns to be in the wrong data type. After converting features back to `float64` type, the model predictions will not be exactly the same again.
